README.md

```diff
@@ -1,115 +1,79 @@
-> describes the status of the Go ecosystem at April 2021.
-
-Currently the official Go compiler cannot produce WebAssembly binaries
-that can be run **outside** of the browser. Because of that, Kubewarden Go
-policies can be built only with the [TinyGo](https://tinygo.org/) compiler.
-
-TinyGo doesn't yet support all the Go features (see [here](https://tinygo.org/lang-support/)
-to see the current project status). Currently its biggest limitation
-is the lack of a fully supported `reflect` package. Among other things, that
-leads to the inability to use the `encoding/json` package against structures
-and user defined types.
+## Settings
 
-Kubewarden policies need to process JSON data like the policy settings and
-the actual request received by Kubernetes.
-However it's still possible to write a Kubewarden policy by using some 3rd party
-libraries.
+The following settings are accepted:
 
-This is a list of libraries that can be useful when writing a Kubewarden
-policy:
+* `forbiddenSysctls`: List of plain sysctl names or sysctl patterns (which end
+  with `*`) to be forbidden. You can forbid a combination of safe and unsafe
+  sysctls in the list. To forbid setting any sysctls, use * on its own.
+* `allowedUnsafeSysctls`: List of plain sysctl names that can be used in Pods.
+  `*` cannot be used.
 
-* Parsing JSON: queries against JSON documents can be written using the
-  [gjson](https://github.com/tidwall/gjson) library. The library features a
-  powerful query language that allows quick navigation of JSON documents and
-  data retrieval.
-* Mutating JSON: changing the contents of a JSON document can be done using the
-  [sjson](https://github.com/tidwall/sjson) library.
-* Generic `set` implementation: using [Set](https://en.wikipedia.org/wiki/Set_(abstract_data_type))
-  data types can significantly reduce the amount of code inside of a policy,
-  see the `union`, `intersection`, `difference`,... operations provided
-  by a Set implementation.
-  The [mapset](https://github.com/deckarep/golang-set) can be used when writing
-  policies.
+A sysctl cannot be both forbidden and allowed at the same time.
 
-Last but not least, this policy takes advantage of helper functions provided
-by [Kubewarden's Go SDK](https://github.com/kubewarden/policy-sdk-go).
+### Example
 
-## Testing
+With this policy deployed and configured as:
 
-This policy comes with a set of unit tests implemented using the Go testing
-framework.
-
-As usual, the tests are defined inside of the `_test.go` files. Given these
-tests are not part of the final WebAssembly binary, the official Go compiler
-can be used to run them. Hence they can take advantage of the `encoding/json`
-package to reduce some testing boiler plate.
-
-The unit tests can be run via a simple command:
-
-```shell
-make test
+```yaml
+forbiddenSysctls:
+- kernel.*
+- net.ipv6.conf.lo.max_addresses
+allowedUnsafeSysctls:
+- net.core.somaxconn
```

Comment:
I think it would be interesting to add an `allowedUnsafeSysctls` more specific than a more generic in `forbiddenSysctls` to show that it has precedence. Like:

```yaml
forbiddenSysctls:
- net.ipv6.conf.lo.*
allowedUnsafeSysctls:
- net.ipv6.conf.lo.max_addresses
```

This instance would prevent every sysctl under `net.ipv6.conf.lo`, except `net.ipv6.conf.lo.max_addresses`, that would be allowed. There might be more interesting examples than this made-up one :)

Reply:
I think we should also add a test case that accepts in this situation. A pod with `sysctl` `net.ipv6.conf.lo.max_addresses` in this context should be allowed, despite `net.ipv6.conf.lo.*` is forbidden. This would test an interesting edge case we want to preserve.

Reply:
Changed unit and e2e tests to check for precedence in https://github.com/kubewarden/psp-sysctl-policy/pull/1/commits/ca24522f6bae4d377951496788d23b4475bd285e.
Changed readme in https://github.com/kubewarden/psp-sysctl-policy/pull/1/commits/ae703568bd79eada02a87e9f7350222ccbc6a446.